### DISCUSSION TITLE
Default system metrics?

### DISCUSSION THREAD

**Discussion by dominikjeske** [open-telemetry/opentelemetry-dotnet#3079] [Q&A]
We are tying to use OpenTelemetry for default metrics like CPU usage but we have 0 results and exception like here https://github.com/open-telemetry/opentelemetry-dotnet/issues/2970. Is there any way to enable standard .NET Core mertics to be visible?

**Comment by dominikjeske** [comment]
Never mind - it looks like standard metrics are exposed in old way ;(

**Reply by cijothomas** [reply]
Its not clear what are you trying to achieve... "old way" refers to.. ?
Are you looking for the ability to automaticaly obtain runtime metrics like this https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Runtime ?

**Reply by dominikjeske** [reply]
@cijothomas Sorry for not being specific. To be honest I'm little lost each time when I'm reading metrics vocabulary in .NET :) So referring to https://docs.microsoft.com/en-us/dotnet/core/diagnostics/compare-metric-apis (existence of this article proves my confusion ;)
- System.Diagnostics.Metrics = new way
- EventCounters = old way

If I understand right OpenTelemetry is reading from System.Diagnostics.Metrics and when we don't have any app metrics it throws exception in current version https://github.com/open-telemetry/opentelemetry-dotnet/issues/2970. I was looking for counters available in https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-counters#dotnet-counters-list but they are only available in EventCounters (**especially I'm looking for Microsoft.AspNetCore.Hosting**).

By saying that if I understand right this https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Runtime fills the gap I'm referring to?

Another sad thing is that dotnet monitor cannot read System.Diagnostics.Metrics (https://github.com/dotnet/dotnet-monitor/issues/329) and only works with EventCounters. I hope this will be resolved because it a little shame

**Reply by cijothomas** [reply]
https://docs.microsoft.com/en-us/dotnet/core/diagnostics/metrics-collection#view-metrics-with-dotnet-counters DotNet Counters surely support the new Metrics API.

> By saying that if I understand right this https://github.com/open-telemetry/opentelemetry-dotnet-contrib/tree/main/src/OpenTelemetry.Instrumentation.Runtime fills the gap I'm referring to?

^ Yes to this.
